src/main/kotlin/org/veupathdb/lib/blast/common/fields/AutoCPUCores.kt

```diff
@@ -0,0 +1,47 @@
+package org.veupathdb.lib.blast.common.fields
+
+import com.fasterxml.jackson.databind.node.ObjectNode
+import org.veupathdb.lib.blast.common.FlagNumThreads
+import org.veupathdb.lib.blast.serial.BlastField
+import org.veupathdb.lib.blast.util.*
+import org.veupathdb.lib.blast.util.add
+import org.veupathdb.lib.blast.util.append
+import org.veupathdb.lib.blast.util.put
+
+private val Def = AutoCPUCoresValue.Disable
+
+internal fun ParseAutoCPUCores(js: ObjectNode) =
+  js.optInt(FlagNumThreads) { AutoCPUCores(parse(it)) } ?: AutoCPUCores()
+
+/**
+ * -num_threads <Integer, >=0>
+ *
+ * Number of threads to use in RPS BLAST search:
+ * * 0 (auto = num of databases)
+ * * 1 (disable)
+ *
+ * Max number of threads = num of databases
+ *
+ * Default = `1`
+ */
+@JvmInline
+value class AutoCPUCores(val value: AutoCPUCoresValue = Def) : BlastField {
+  override val isDefault get() = value == Def
+
+  override fun appendJson(js: ObjectNode) =
+    js.put(isDefault, FlagNumThreads, value.ordinal)
```

Comment:
Maybe a nitpick, is it worthwhile to give this enum a member value distinct from ordinal so that the flag value is not coupled to the order? It probably doesn't matter much especially since the enum is defined in this class and is only used here.

Reply:
A valid nitpick, I personally like using the ordinal simply to not have to hold onto 2 more integers in memory, but that is like the epitome of needless micro-optimizations.

Reply:
To be perfectly honest, I don't like using enums in the first place due to the jvm class/reflection overhead.  But again, needless micro-optimizations.